### ISSUE TITLE
Segmentation fault in KDE Neon 5.11.4

### ISSUE THREAD

**Issue by manulaiko** [Kazakuri/Shinjiru#66]
I've followed the install instructions and everything seems correct.
When I run the executable at `.local/bin/Shinjiru` I get a segfault when *Authorization Code Request* window appears.

Output of install script (without compiler):

```
┌─[manulaiko@manulaiko-Lenovo-B50-10]─[~]
└──╼ $ ./shinjiru.sh 
fatal: destination path 'Shinjiru' already exists and is not an empty directory.
Already up-to-date.
make: Nothing to be done for 'first'.
/home/manulaiko/Shinjiru/lib/fervor/quazip/quazipfile.cpp:217: Cannot invoke tr() like this
/home/manulaiko/Shinjiru/lib/fervor/quazip/quazipfile.cpp:217: Class 'QuaZipFilePrivate' lacks Q_OBJECT macro
/home/manulaiko/Shinjiru/src/lib/windowwatcher.cpp:119: Unterminated C++ string
/home/manulaiko/Shinjiru/src/lib/windowwatcher.cpp:121: Excess closing parenthesis in C++ code (or abuse of the C++ preprocessor)
/home/manulaiko/Shinjiru/src/lib/windowwatcher.cpp:121: Unterminated C++ string
Updating 'fervor_lt.ts'...
    Found 331 source text(s) (0 new and 331 already existing)
Updating 'shinjiru_it.ts'...
    Found 331 source text(s) (0 new and 331 already existing)
Updating 'shinjiru_de.ts'...
    Found 331 source text(s) (0 new and 331 already existing)
Updating 'shinjiru_es.ts'...
    Found 331 source text(s) (0 new and 331 already existing)
Updating 'shinjiru_ja.ts'...
    Found 331 source text(s) (0 new and 331 already existing)
Updating 'shinjiru_pt.ts'...
    Found 331 source text(s) (0 new and 331 already existing)
Updating '/home/manulaiko/Shinjiru/fervor_lt.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 331 untranslated source text(s)
Updating '/home/manulaiko/Shinjiru/shinjiru_it.qm'...
    Generated 327 translation(s) (312 finished and 15 unfinished)
    Ignored 4 untranslated source text(s)
Updating '/home/manulaiko/Shinjiru/shinjiru_de.qm'...
    Generated 308 translation(s) (308 finished and 0 unfinished)
    Ignored 23 untranslated source text(s)
Updating '/home/manulaiko/Shinjiru/shinjiru_es.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 331 untranslated source text(s)
Updating '/home/manulaiko/Shinjiru/shinjiru_ja.qm'...
    Generated 99 translation(s) (56 finished and 43 unfinished)
    Ignored 232 untranslated source text(s)
Updating '/home/manulaiko/Shinjiru/shinjiru_pt.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 331 untranslated source text(s)
cp: will not overwrite just-created 'data/translations/fervor_lt.qm' with 'lib/fervor/fervor_lt.qm'
/usr/lib/qt5/bin/qmake -install qinstall -exe /home/manulaiko/Shinjiru/Shinjiru /home/manulaiko/.local/bin/Shinjiru
Error copying /home/manulaiko/Shinjiru/Shinjiru to /home/manulaiko/.local/bin/Shinjiru: Destination file exists
Makefile:2475: recipe for target 'install_bin' failed
make: [install_bin] Error 3 (ignored)
strip /home/manulaiko/.local/bin/Shinjiru
strip:/home/manulaiko/.local/bin/Shinjiru: could not create temporary file to hold stripped copy: No error
Makefile:2475: recipe for target 'install_bin' failed
make: [install_bin] Error 1 (ignored)
/usr/lib/qt5/bin/qmake -install qinstall /home/manulaiko/Shinjiru/data/skin /home/manulaiko/.local/share/Kazakuri/Shinjiru/skin
/usr/lib/qt5/bin/qmake -install qinstall /home/manulaiko/Shinjiru/data/translations /home/manulaiko/.local/share/Kazakuri/Shinjiru/translations
Error copying /home/manulaiko/Shinjiru/data/translations/fervor_lt.qm to /home/manulaiko/.local/share/Kazakuri/Shinjiru/translations/fervor_lt.qm: Destination file exists
Makefile:2485: recipe for target 'install_data' failed
make: [install_data] Error 3 (ignored)
install -m 644 -p res/icon.svg /home/manulaiko/.local/share/icons/hicolor/scalable/apps/Shinjiru.svg
cp -f /home/manulaiko/Shinjiru/Shinjiru.desktop.in /home/manulaiko/Shinjiru/Shinjiru.desktop; echo "Exec=/home/manulaiko/.local/bin/Shinjiru" >> /home/manulaiko/Shinjiru/Shinjiru.desktop; install -m 644 -p /home/manulaiko/Shinjiru/Shinjiru.desktop /home/manulaiko/.local/share/applications/Shinjiru.desktop
/usr/lib/qt5/bin/qmake -install qinstall /home/manulaiko/Shinjiru/LICENSE /home/manulaiko/.local/share/licenses/Shinjiru/LICENSE
Error copying /home/manulaiko/Shinjiru/LICENSE to /home/manulaiko/.local/share/licenses/Shinjiru/LICENSE: Destination file exists
Makefile:2514: recipe for target 'install_license' failed
make: [install_license] Error 3 (ignored)

------------------------------------
Shinjiru was successfully installed!

You should now be able to see Shinjiru in the 'Internet' category of your application launcher.
The binary file is located at '/home/manulaiko/.local/bin/Shinjiru'.
┌─[manulaiko@manulaiko-Lenovo-B50-10]─[~]
└──╼ $ .local/bin/Shinjiru 
Segmentation fault
```

Tool versions:
```
┌─[manulaiko@manulaiko-Lenovo-B50-10]─[~]
└──╼ $ qmake --version; gcc -v; lsb_release -a
QMake version 3.1
Using Qt version 5.9.3 in /usr/lib/x86_64-linux-gnu
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/5/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 5.4.0-6ubuntu1~16.04.5' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.5) 
No LSB modules are available.
Distributor ID: neon
Description:    KDE neon User Edition 5.11
Release:        16.04
Codename:       xenial
```

**Comment by Kazakuri** [comment]
Hey, there.

I'm fairly certain the culprit is [this block of code](https://github.com/Kazakuri/Shinjiru/blob/82c1041f63e2c7bf947a11e6e386e430c61e6e63/src/lib/torrentrss.cpp#L98).
When nyaa.si came forth to replace nyaa.se they originally kept the same structure of RSS feed, before changing it without warning one day and breaking the existing RSS parsing. 

I did do a release for Windows that addressed the issue, but apparently I never pushed up the updated code.
My laptop's battery ended up exploding and with the introduction of APIv2 Shinjiru development has sort of shifted into a rewrite using APIv2, but I'll get the fix pushed up in a bit.

**Comment by manulaiko** [comment]
I've made the fix, it now works perfectly but there's no way to see if a torrent is trusted or not.
I'll prepare the PR in a while.

**Comment by Kazakuri** [comment]
Thanks again for the PR.

As far as trusted torrents goes:
When working on the new version of Shinjiru for APIv2 I was planning on having a wrapper around the RSS feed to support extras like pagination and searching (iirc nyaa.si broke that too).
So it might be worth considering following both the default rss feed and the trusted-only rss feed on the wrapper and add in a new tag for trusted/untrusted

**Comment by manulaiko** [comment]
Is there any roadmap for v2? I'd like to help, but idk what must be done.

**Comment by Kazakuri** [comment]
I created a repo for it on a GitLab CE installation on a server of mine [here](https://git.urus.ai/shinjiru/Shinjiru) earlier today.
Up until now I've been going back and forth with ideas about languages and frameworks and went back to Qt after having some issues with JS and Electron.

It's very early on in development right now (since I only started on it again last night) because of all the experimentation with frameworks, so there's not really any sort of roadmap going on right now.

AniList APIv1 is going to be shutting down May 1st so over the next month or so the rewrite should get a lot more flushed out and there can be a more complete image of what stills needs to be done.
